src/matrixeventprocessor.ts

```diff
@@ -212,6 +206,78 @@ export class MatrixEventProcessor {
         return `[${name}](${url})`;
     }
 
+    public async GetEmbedForReply(event: any): Promise<[Discord.RichEmbed, string]|undefined> {
+        const REPLY_MATCHES = 4;
+        const INDEX_REPLY = 3;
+        const INDEX_SOURCE = 2;
+        const INDEX_SENDER = 1;
+        const relatesTo = event.content["m.relates_to"];
+        let eventId = null;
+        if (relatesTo && relatesTo["m.in_reply_to"]) {
+            eventId = relatesTo["m.in_reply_to"].event_id;
+        } else {
+            return;
+        }
+        const matches = REPLY_REGEX.exec(event.content.body);
+        if (!matches || matches.length !== REPLY_MATCHES) {
+            return;
+        }
+        const intent = this.bridge.getIntent();
+        const embed = new Discord.RichEmbed();
+        // Try to get the event.
+        try {
+            const sourceEvent = await intent.getEvent(eventId);
+            let replyText = sourceEvent.content.body  || "Reply with unknown content";
+            // Check if this is also a reply.
+            if (sourceEvent.content && sourceEvent.content["m.relates_to"] &&
+                sourceEvent.content["m.relates_to"]["m.in_reply_to"]) {
+                const sourceMatch = REPLY_REGEX.exec(sourceEvent.content.body);
+                if (sourceMatch && sourceMatch.length === REPLY_MATCHES) {
+                    replyText = sourceMatch[INDEX_REPLY];
```

Comment:
message 1: <stuff>
message 2: <reply to message 1>
message 3: <reply to message 2>

we are replying to message 2, though, and not message 1, so why bother to get that source?